test/functional/Valhalla/src_qtypes/org/openj9/test/lworld/ValueTypeUnsafeTests.java

```diff
@@ -307,65 +322,87 @@ static public void testNonVTClzGetValue() throws Throwable {
 		assertNull(myUnsafe.getValue(vtPoint, intWrapperOffsetVti, IntWrapper.class));
 	}
 
+	@NullRestricted
+	static ValueTypePoint2D testGetValuesOfArray_p;
+
 	@Test
 	static public void testGetValuesOfArray() throws Throwable {
 		if (isFlatteningEnabled) {
-			ValueTypePoint2D p = myUnsafe.getValue(vtPointAry, vtPointAryOffset0, ValueTypePoint2D.class);
-			assertEquals(p.x.i, vtPointAry[0].x.i);
-			assertEquals(p.y.i, vtPointAry[0].y.i);
-			p = myUnsafe.getValue(vtPointAry, vtPointAryOffset1, ValueTypePoint2D.class);
-			assertEquals(p.x.i, vtPointAry[1].x.i);
-			assertEquals(p.y.i, vtPointAry[1].y.i);
+			testGetValuesOfArray_p = myUnsafe.getValue(vtPointAry, vtPointAryOffset0, ValueTypePoint2D.class);
+			assertEquals(testGetValuesOfArray_p.x.i, vtPointAry[0].x.i);
+			assertEquals(testGetValuesOfArray_p.y.i, vtPointAry[0].y.i);
+			testGetValuesOfArray_p = myUnsafe.getValue(vtPointAry, vtPointAryOffset1, ValueTypePoint2D.class);
+			assertEquals(testGetValuesOfArray_p.x.i, vtPointAry[1].x.i);
+			assertEquals(testGetValuesOfArray_p.y.i, vtPointAry[1].y.i);
 		}
 	}
 
 	@Test
 	static public void testGetValueOfZeroSizeVTArrayDoesNotCauseError() throws Throwable {
-		ZeroSizeValueType[] zsvtAry = new ZeroSizeValueType[] {
-			new ZeroSizeValueType(),
-			new ZeroSizeValueType()
-		};
+		ZeroSizeValueType[] zsvtAry = (ZeroSizeValueType[])ValueClass.newArrayInstance(
+			NullRestrictedCheckedType.of(ZeroSizeValueType.class), 2);
+		zsvtAry[0] = new ZeroSizeValueType();
+		zsvtAry[1] = new ZeroSizeValueType();
 		long zsvtAryOffset0 = myUnsafe.arrayBaseOffset(zsvtAry.getClass());
 		assertNotNull(myUnsafe.getValue(zsvtAry, zsvtAryOffset0, ZeroSizeValueType.class));
 	}
 
+	@NullRestricted
+	static ZeroSizeValueTypeWrapper zsvtw;
```

Comment:
We should keep this as local variable, and put the `Unsafe.getValue()` inside `if (Unsafe.isFieldAtOffsetFlattened()) { }`. 

Similarly for the `Unsafe.getValue(), Unsafte.putValue()` test cases in this file, they can be kept as local variables. We just need to add additional check like `Unsafe.isFlatArray()`,` Unsafe.isFlatField()` or `Unsafe.isFieldAtOffsetFlattened()` before them.

Reply:
Discussed on slack, we can keep these variables as locals (without changing them to null restricted static) if the tests are already passing.